dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -118,19 +118,67 @@ There is some ambiguity between property, transform, and computation, so use you
 
 ### H3 Grid Functions
 
+
+#### Distance
+
 | Current name |      Proposed name      |     Notes      |
 |--------------|-------------------------|----------------|
 | `h3Distance` | `gridDistance`          |                |
 | `h3Line`     | `gridPathCells`         |                |
 | *DNE*        | `gridPathEdges`         |                |
 | *DNE*        | `gridPathDirectedEdges` |                |
-| `kRing`      | `gridDisk`              | filled-in disk |
-| `hexRing`    | `gridRing`              | hollow ring    |
 
-todo: add C function to correspond to "works with pentagons" version of `hexRing`/`gridRing`
 
-todo: For the `kRing`, `hexRange`, `hexRing`, etc. family of functions, should we come up with some standard prefix or suffix to denote that the function will fail if it encounters a pentagon?
+#### Filled-In Disk With Distances
+
+|     Current name    |      Proposed name       |    Type   |     Notes      |                Calls                 |
+|---------------------|--------------------------|-----------|----------------|--------------------------------------|
+| `hexRangeDistances` | `gridDiskDistances_fast` | hex only  | disk/distances | NONE                                 |
+| `_kRingInternal`    | `gridDiskDistances_slow` | pentagons | disk/distances | NONE                                 |
+| `kRingDistances`    | `gridDiskDistances`      | general   | disk/distances | `hexRangeDistances`,`_kRingInternal` |
+
+#### Filled-In Disk Without Distances
+
+| Current name |  Proposed name   |    Type   | Notes |                Calls                 |
+|--------------|------------------|-----------|-------|--------------------------------------|
+| `kRing`      | `gridDisk`       | general   | disk  | `kRingDistances`, drops distances    |
+| `hexRange`   | `gridDisk_fast`? | hex only  | disk  | `hexRangeDistances`, drops distances |
+| *DNE*        | `gridDisk_slow`? | pentagons | disk  |                                      |
+
+#### Hollow Ring
+
+| Current name |  Proposed name  |    Type   | Notes | Calls |
+|--------------|-----------------|-----------|-------|-------|
+| *DNE*        | `gridRing`      | general   | ring  |       |
+| `hexRing`    | `gridRing_fast` | hex only  | ring  | NONE  |
+| *DNE*        | `gridRing_slow` | pentagons | ring  |       |
+
+
+#### To Remove
+
+| Current name | Proposed name |   Type   |     Calls      |
+|--------------|---------------|----------|----------------|
+| `hexRanges`  | (remove)      | hex only | N x `hexRange` |
+
+
+#### Todo
+- maybe there's a better suffix than `_fast`/`_slow`?
+- what's going on with these distance allocations (or lack of)?
+
+```C
+// unnecessary malloc?
```

Comment:
Not unnecessary, if you look at the `kRingDistances` implementation it needs the "scratch space" to work.

Reply:
Interesting. So `kRingDistances` depends on `hexRangeDistances` and `_kRingInternal`. From what I can tell, only `_kRingInternal` needs the "scratch space", but `hexRangeDistances` does not.

That means the `malloc` for `kRingDistances` is not needed for the "fast path" that only calls `hexRangeDistances`. If that's the majority of cases, should we re-factor this code to skip the `malloc`?

Reply:
if we push the `malloc` into `kRingDistances` or `_kRingInternal` but I'd need to look at the function signatures to see if that's even possible (is the size value available at that point in the call stack?)

Reply:
Instead of pushing it in to those functions, we could pull it out into `kRing`: #319